profiles/admin.py

```diff
@@ -197,7 +204,8 @@ def upload_json(self, request):
         try:
             if request.method == "POST":
                 data = json.loads(request.FILES["json_file"].read())
-                result = Profile.import_customer_data(data)
+                service_name = request.POST["service_name"]
```

Comment:
Now that there is a non-file field in the form, the form handling could be done like it's usually done with Django forms.

i.e. instantiating the form, calling is_valid and using form.cleaned_data.

Reply:
Sounds like a good direction. I'm not that familiar with Django forms, so have to study a bit what is it exactly that should be done :monocle_face:

Reply:
[Here](https://docs.djangoproject.com/en/2.2/topics/forms/#the-view) is an example in the Django docs.

Reply:
Actually, I started thinking, that my time is probably better spent doing something else. While this would be a good change, perhaps it's not a good time to do it now. I don't even know whether this form has any use (any more). So I'm suggesting not to change this right now.

Reply:
It shouldn't take too long. Something like this:

```python
    @method_decorator(superuser_required, name="dispatch")
    def upload_json(self, request):
        try:
            if request.method == "POST":
                form = ImportProfilesFromJsonForm(request.POST, request.FILES)
                if form.is_valid():
                    data = json.loads(request.FILES["json_file"].read())
                    service_name = form.cleaned_data["service_name"]
                    result = Profile.import_customer_data(data, service_name)
                    response = JsonResponse(result)
                    response["Content-Disposition"] = "attachment; filename=export.json"

                    return response
                else:
                    raise ValidationError(', '.join(form.errors.values()))
            else:
                form = ImportProfilesFromJsonForm()
                return render(request, "admin/profiles/upload_json.html", {
                    "form": form
                })
        except Exception as err:
            messages.error(request, err)
            form = ImportProfilesFromJsonForm()
            return render(request, "admin/profiles/upload_json.html", {
                "form": form
            })
```

Reply:
Allrighty, not that big a change, so bit the bullet. The form conveniently gives the actual `Service` directly, so no need to fetch it explicitly.